feat: переход на базу данных H2 (Спринт 11)

DanSBol opened the pull request:
переход на базу данных H2 (Спринт 11)

DanSBol committed:
feat: переход на базу данных H2 (Спринт 11)

DanSBol committed:
Update Film.java

DanSBol committed:
Merge branch 'main' into add-database

DanSBol committed:
Update User.java

DanSBol committed:
Update FilmDbStorage.java

DanSBol committed:
Update UserDbStorage.java

DanSBol committed:
Update UserDbStorage.java

DanSBol committed:
Update UserDbStorage.java

DanSBol committed:
Update UserDbStorage.java

DanSBol committed:
Update UserDbStorage.java

DanSBol committed:
Update UserDbStorage.java

DanSBol committed:
Update UserDbStorage.java

DanSBol committed:
Update FilmDbStorageTest.java

DanSBol committed:
Update User.java

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -4,59 +4,82 @@
     private final FilmService filmService;
 
-    @GetMapping("/{id}")
+    @GetMapping("/genres")
```
в отдельный контроллер GenreController

DanSBol replied:
Добавил отдельный контроллер и, дополнительно, к нему сервис GenreService, интерфейс GenreStorage и реализацию GenreDbStorage

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -4,59 +4,82 @@
+        return filmService.getGenre(id);
+    }
+
+    @GetMapping("/mpa")
```
в отдельный контроллер MpaController

DanSBol replied:
Добавил отдельный контроллер и, дополнительно, к нему сервис MpaService, интерфейс MpaStorage и реализацию MpaDbStorage

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -1,65 +1,52 @@
     private Set<Integer> likes;
 
-    public Film(String name, String description, LocalDate releaseDate, int duration, Genre genre, Rating rating) {
+    public Film(int id, String name, String description, LocalDate releaseDate, int duration, Set<Genre> genres,
```
зачем такие большие конструкторы? это очень не удобно, попробуйте @Builder

DanSBol replied:
Поменял работу с конструктором на работу с @builder. Мельком прошли, я отразил, как прикольно, но не нашел как добавить. Сейчас добавил. Спс

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/model/Rating.java:
```diff
@@ -1,9 +1,15 @@
 package ru.yandex.practicum.filmorate.model;
 
-public enum Rating {
```
вы можете вернуть этот enum, только назвать RatingEnum (или как по другому)
и использовать вместо String name RatingEnum name;

@Enumerated(String.class) -- вроде так
private RatingEnum name;

DanSBol replied:
Попробовал...

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -5,69 +5,88 @@
     @Autowired
-    public FilmService(InMemoryFilmStorage inMemoryFilmStorage) {
-        this.inMemoryFilmStorage = inMemoryFilmStorage;
+    public FilmService(FilmStorage filmStorage) {
```
@RequiredArgsConstructor  вместо конструктора

DanSBol replied:
Поправил

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -6,73 +6,67 @@
     @Autowired
-    public UserService(InMemoryUserStorage inMemoryUserStorage) {
-        this.inMemoryUserStorage = inMemoryUserStorage;
+    public UserService(UserStorage userStorage) {
```
@RequiredArgsConstructor  вместо конструктора

DanSBol replied:
Поправил

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/storage/FilmDbStorage.java:
```diff
@@ -0,0 +1,247 @@
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public FilmDbStorage(JdbcTemplate jdbcTemplate) {
```
@RequiredArgsConstructor  вместо конструктора

DanSBol replied:
Поправил

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/storage/FilmDbStorage.java:
```diff
@@ -0,0 +1,247 @@
+    }
+
+    @Override
+    public Optional<Film> get(int id) {
```
getFilm

DanSBol replied:
Переименовал все методы на более понятные

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/storage/FilmDbStorage.java:
```diff
@@ -0,0 +1,247 @@
+    }
+
+    @Override
+    public Collection<Film> getAll() {
```
getAllFilms

DanSBol replied:
Переименовал все методы на более понятные

ErkinKasymaliev reviewed (changes requested):
привет, оставил комменты

DanSBol committed:
fix: устранить замечания ревьювера (Спринт 11)
- использовать @builder вместо громоздких конструкторов
- использовать @RequiredArgsConstructor вместо конструкторов со всеми параметрами
- вернуть enum применительно к наименованию mpa
- работу с Genre и Mpa вынести в отдельные контроллеры

DanSBol committed:
Merge remote-tracking branch 'origin/add-database' into add-database

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/model/Film.java
#	src/main/java/ru/yandex/practicum/filmorate/model/User.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/FilmDbStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/UserDbStorage.java

ErkinKasymaliev reviewed (approved):
привет, работа принята
